### ISSUE TITLE
Recommended Viscosity Method for Highly Varying Physical Viscosity

### ISSUE THREAD

**Issue by kyletreece** [InteractiveComputerGraphics/SPlisHSPlasH#97]
Hello! I recently downloaded and started testing this repository and am very impressed so far and I'm excited to continue using it! I'm interested in making some updates to enable running simulations of melting, and was wondering if you would recommend a specific viscosity method to use as a baseline. I would like to simulate things such as an ice cube melting, so starting with extremely high viscosity (solid) and ending with extremely low viscosity (fully liquid flow).

Reading the viscosity method 7 paper (Weiler et al. 2018) makes me think this will be the best option, especially since the authors mentioned they were able to configure a temperature dependent viscosity affect that uses the actual physically real dynamic viscosity. But I do notice the objects and materials simulated in this paper all have significantly higher viscosity values than what I'm interested in simulating.

When testing viscosity method 7 in the basic DamBreakModel case I'm noticing what appears to be excessive "spattering" or instabilities when trying to simulate water using a viscosity of 0.001 and density of 1000. This is at least in comparison to using viscosity method 1, which runs extremely smoothly and looks more convincing.

Thank you in advance for your advice and your time!

**Comment by janbender** [comment]
I would also recommend the method of Weiler. In theory this should also work for small coefficients. I'll have to test why this is not working correctly in the new year.



Am 17. Dezember 2020 23:59:01 MEZ schrieb kyletreece <notifications@github.com>:

**Comment by kyletreece** [comment]
Hi @janbender, thank you for your reply and happy new year! I reviewed the results that prompted my original question, and I suspect they may actually make more sense than I originally thought. The "spattering" I observed makes more intuitive sense when considering the actual scale of the case (the dam break case is actually a LOT of water), and that I had not enabled any of the surface tension implementations.
